exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -234,8 +238,32 @@ static MetricDescriptor createMetricDescriptor(View view, String projectId) {
     return builder.build();
   }
 
-  private static String generateType(String viewName) {
-    return CUSTOM_OPENCENSUS_DOMAIN + viewName;
+  private static String generateType(
+      String viewName, @javax.annotation.Nullable String metricNamePrefix) {
+    String domain;
+    if (metricNamePrefix == null) {
```

Comment:
Can we cache this logic? I mean the calculation of the domain. Is this does not happen for all the points then we can just add a comment to explain that.